barretenberg/ts/src/proof/index.ts

```diff
@@ -46,12 +44,10 @@ export function splitHonkProof(
 }
 
 export function reconstructHonkProof(publicInputs: Uint8Array, proof: Uint8Array): Uint8Array {
-  const proofStart = proof.slice(0, serializedBufferSize);
-  const proofEnd = proof.slice(serializedBufferSize);
-
-  // Concatenate publicInputs and proof
-  const proofWithPublicInputs = Uint8Array.from([...proofStart, ...publicInputs, ...proofEnd]);
+  // Append the buffer length to the concatenated proof
+  const bufferLength = numToUInt32BE((publicInputs.length + proof.length) / fieldByteSize);
```

Comment:
publicInputs.length + proof.length is already the size in bytes

is bufferLength measured in number of field elements or number of bytes?

Reply:
its the number of field elements - i checked the value of the first 4 bytes of the proof we have now
i should rename the variable `bufferLength`